### DISCUSSION TITLE
qdrant vector store is not working - 'NoneType' object has no attribute 'collection_exists'

### DISCUSSION THREAD

**Discussion by forcewake** [open-webui/open-webui#6386] [General]
# Bug Report

Looks like new `QdrantClient` doesn't work properly - I've got `'NoneType' object has no attribute 'collection_exists'` each time I've tried to upload file (from chat ui, from knowledge base and etc)

## Installation Method

docker

## Environment

- **Open WebUI Version:** [v0.3.33]
- **Ollama (if applicable):** [v0.3.14]

- **Operating System:** [Windows 11]
- **Browser (if applicable):** [Edge Version 130.0.2849.46 (Official build) (64-bit)]

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

[Describe what you expected to happen.]

## Actual Behavior:

qdrant is up and running

I've got `'NoneType' object has no attribute 'collection_exists'` each time I've tried to upload file (from chat ui, from knowledge base and etc).

## Description

**Bug Summary:**
[Provide a brief but clear summary of the bug]

## Reproduction Details

**Steps to Reproduce:**
[Outline the steps to reproduce the bug. Be as detailed as possible.]

## Logs and Screenshots

**Browser Console Logs:**
```
{id: 'ad6c21eb-74b3-4446-9017-be5a526672ad', user_id: 'f1d07687-363e-45b4-9531-a6eceaac6d1f', hash: null, filename: 'ad6c21eb-74b3-4446-9017-be5a526672ad_1.THE_RED_BULL_RACING (1).pdf', path: '/app/backend/data/uploads/ad6c21eb-74b3-4446-9017-be5a526672ad_1.THE_RED_BULL_RACING (1).pdf', …}
index.ts:144 
        
        
        POST https://xx/api/v1/knowledge/c54bdbe9-7f3c-4b6e-a2df-d697774e3eb6/file/add 400 (Bad Request)
window.fetch @ fetcher.js:76
u @ index.ts:144
ke @ Collection.svelte:356
g @ Collection.svelte:165
await in g
Me @ Collection.svelte:584
Show 1 more frame
Show less
[NEW] Explain Console errors by using Copilot in Edge: click
         
         to explain an error. 
        Learn more
        Don't show again
index.ts:165 {detail: "400: 'NoneType' object has no attribute 'collection_exists'"}
```

**Docker Container Logs:**
```
IINFO  [open_webui.apps.retrieval.main] save_docs_to_vector_db [Document(metadata={'Content-Type': 'application/pdf'}, page_content='\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n')] file-ad6c21eb-74b3-4446-9017-be5a526672ad
ERROR [open_webui.apps.retrieval.main] 'NoneType' object has no attribute 'collection_exists'
Traceback (most recent call last):
  File "/app/backend/open_webui/apps/retrieval/main.py", line 888, in process_file
    raise e
  File "/app/backend/open_webui/apps/retrieval/main.py", line 862, in process_file
    result = save_docs_to_vector_db(
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backend/open_webui/apps/retrieval/main.py", line 650, in save_docs_to_vector_db
    result = VECTOR_DB_CLIENT.query(
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backend/open_webui/apps/retrieval/vector/dbs/qdrant.py", line 97, in query
    if not self.has_collection(collection_name):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backend/open_webui/apps/retrieval/vector/dbs/qdrant.py", line 66, in has_collection
    return self.client.collection_exists(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'collection_exists'
ERROR [open_webui.apps.webui.routers.files] 400: 'NoneType' object has no attribute 'collection_exists'
Traceback (most recent call last):
  File "/app/backend/open_webui/apps/retrieval/main.py", line 888, in process_file
    raise e
  File "/app/backend/open_webui/apps/retrieval/main.py", line 862, in process_file
    result = save_docs_to_vector_db(
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backend/open_webui/apps/retrieval/main.py", line 650, in save_docs_to_vector_db
    result = VECTOR_DB_CLIENT.query(
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backend/open_webui/apps/retrieval/vector/dbs/qdrant.py", line 97, in query
    if not self.has_collection(collection_name):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backend/open_webui/apps/retrieval/vector/dbs/qdrant.py", line 66, in has_collection
    return self.client.collection_exists(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'collection_exists'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/app/backend/open_webui/apps/webui/routers/files.py", line 71, in upload_file
    process_file(ProcessFileForm(file_id=id))
  File "/app/backend/open_webui/apps/retrieval/main.py", line 897, in process_file
    raise HTTPException(
fastapi.exceptions.HTTPException: 400: 'NoneType' object has no attribute 'collection_exists'
ERROR [open_webui.apps.webui.routers.files] Error processing file: ad6c21eb-74b3-4446-9017-be5a526672ad
INFO:     192.168.1.1:0 - "POST /api/v1/files/ HTTP/1.1" 200 OK
ERROR [open_webui.apps.retrieval.main] 'NoneType' object has no attribute 'collection_exists'
Traceback (most recent call last):
  File "/app/backend/open_webui/apps/retrieval/main.py", line 798, in process_file
    result = VECTOR_DB_CLIENT.query(
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backend/open_webui/apps/retrieval/vector/dbs/qdrant.py", line 97, in query
    if not self.has_collection(collection_name):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backend/open_webui/apps/retrieval/vector/dbs/qdrant.py", line 66, in has_collection
    return self.client.collection_exists(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'collection_exists'
INFO:     192.168.1.1:0 - "POST /api/v1/knowledge/c54bdbe9-7f3c-4b6e-a2df-d697774e3eb6/file/add HTTP/1.1" 400 Bad Request
```

**Screenshots/Screen Recordings (if applicable):**

## Additional Information

I know that it is new feature and was not share public yet but it's already in main branch.

**Comment by tjbck** [comment]
Qdrant support is not supported officially.

Added by @RobinBially

**Reply by enving** [reply]
is qdrant not officially supported still? I get the same error... but in openwebui docs it is mentioned as a vectordb... im a bit lost :D

**Reply by neilor-mendes** [reply]
Oh man... What is supported then?

**Reply by madsit83** [reply]
well, even if not supported, it used to work fine, I am also getting this error on a clean open-webui install through docker, qdrant is running fine and reachable from the container, but the 400 error persists no matter what I do.

**Comment by tdiprima** [comment]
Hey folks — just a heads-up that I'm hitting the same error:  
**`400: 'NoneType' object has no attribute 'collection_exists'`**

It pops up when trying to add documents using Qdrant as the vector store. I've confirmed Qdrant is running and accessible (even tested it directly via the Python client — inserts and searches work fine), so it seems like something's broken in the WebUI-Qdrant integration layer.

Honestly, I *really* want to use Open WebUI for a production-ish RAG setup, but it's been tough to get it working smoothly. **Even with just 100–200 docs, RAG is painfully slow**, and the integration with Qdrant just isn't behaving as expected. I know it's not supported, but maybe it *oughta be*?  It'd be awesome if the team could take a look at:

* Proper Qdrant connection handling
* Better error logging for RAG workflows
* Some way to verify that embeddings are stored and retrieved correctly

I know this project is moving fast (which is awesome), but **RAG is such a core use case right now that getting this solid would really level it up.**

Happy to help test things if needed. Thanks for all the work you're doing — just hoping this part gets a little more love. 🙏

---

### UPDATE: RESOLVED [#15197](https://github.com/open-webui/open-webui/issues/15197#issuecomment-3013550832)

**Comment by kaiiquetome** [comment]
No meu caso, estava obtendo o mesmo erro conforme log abaixo. Eu estava passando a variavel de ambiente errada, quando na verdade o método do qdrant faz a leitura da variável QDRANT_URI.

2025-06-30 18:23:15.796 | ERROR    | open_webui.routers.retrieval:process_file:1457 - 'NoneType' object has no attribute 'collection_exists' - {}
 
variavel de ambiente
      - VECTOR_DB=qdrant
      - QDRANT_HOST=http://qdrant:6333
      - QDRANT_URI=http://qdrant:6333  <-- linha que resolveu o problema
      - ENV=dev
      - LOG_LEVEL=DEBUG
      - OFFLINE_MODE=false
